docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -20,196 +45,193 @@ The scenarios of defining placement rules in SQL include:
-	...
-```
+* rule_id
+* target ID
+* target name
+* constraints
+* role
+* replicas
+* scheduling state
 
-This statement indicates TiDB to overwrite the current placement rule with the same `role`. It affects all data of table `table_name`, including indices.
+The system table is a virtual table, which doesn’t persist data. When querying the table, TiDB queries PD and integrates the result in a table format. That also means the metadata is stored on PD instead of TiKV.
 
-Assuming table `test` has 3 replicas by default, the default placement rule is equivalent to:
+An object may contain multiple placement rules, each of which corresponds to a rule in PD.
 
-```sql
-ALTER TABLE test
-	ADD PLACEMENT POLICY ROLE=voter REPLICAS=3;
-```
+Advantages of building system table include:
 
-`CONSTRAINTS` is omitted here, because there is no label constraints on voters.
+* It’s easy for users to filter and aggregate the result
+* There’s no need to support a new grammar, and it’s easier to implement
 
-Since at most one rule can be defined for each role, `ALTER PLACEMENT POLICY` will replace the existing rule with the same role. For example:
+#### SHOW PLACEMENT
 
-```sql
-ALTER TABLE test
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=voter REPLICAS=2,
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=bj]" ROLE=voter REPLICAS=2,
-	ALTER PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=voter REPLICAS=3,
-	ALTER PLACEMENT POLICY CONSTRAINTS="[+zone=bj]" ROLE=voter REPLICAS=3;
-```
+The system table only contains stored placement rules, and users cannot query the effective rule of one object from it.
 
-As all the rules are defined on the same role `voter`, the first 3 rules will be overwritten by the last one. So it is equivalent to:
+For example, table `t` has two partitions `p0` and `p1`, and a placement rule is added on `t`. If the user wants to query the working rule of `p0`, he will find no placement rule is defined for `p0` through the system table. Based on the inheritance rules for partitioned tables the user needs to query the placement rule on `t`. This procedure is annoying.
 
-```sql
-ALTER TABLE test
-	ALTER PLACEMENT POLICY CONSTRAINTS="[+zone=bj]" ROLE=voter REPLICAS=3;
-```
+To simplify the procedure, a `SHOW PLACEMENT` statement is provided to summarize the effective rules for one specified object.
 
-To add a prohibiting constraint to all the placement rules can be only achieved by overwriting all the rules. For example, assuming the original placement rules are:
+The statement is in such a format:
 
 ```sql
-ALTER TABLE test
-	ALTER PLACEMENT POLICY CONSTRAINTS="[+zone=bj]" ROLE=voter REPLICAS=3;
-	ALTER PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=follower REPLICAS=2;
+SHOW PLACEMENT FOR {DATABASE | SCHEMA} schema_name;
+SHOW PLACEMENT FOR TABLE table_name [PARTITION partition_name];
+SHOW PLACEMENT FOR INDEX index_name FROM table_name [PARTITION partition_name];
```

Comment:
I've been removing INDEX from the proposal since it is not useful for now. It does have a very clear use-case with global indexes for partitioned tables however.